lib/AbiBinProvider.js

```diff
@@ -22,18 +21,32 @@ class OpenSTAbiBinProvider extends AbiBinProvider {
   /**
    * Constructor for OpenSTAbiBinProvider.
    *
-   * @param abiFolderPath Folder path where abi is present.
-   * @param binFolderPath Folder path where bin is present.
-   * @param mosaicAbiFolderPath Folder path where mosaic abi is present.
-   * @param mosaicBinFolderPath Folder path where mosaic bin is present.
+   * @param {string} [abiFolderPath] Folder path where abi is present.
+   * @param {string} [binFolderPath] Folder path where bin is present.
    */
-  constructor(abiFolderPath, binFolderPath, mosaicAbiFolderPath, mosaicBinFolderPath) {
-    abiFolderPath = abiFolderPath || DEFAULT_ABI_FOLDER_PATH;
-    binFolderPath = binFolderPath || DEFAULT_BIN_FOLDER_PATH;
-    super(abiFolderPath, binFolderPath);
+  constructor(abiFolderPath, binFolderPath) {
+    const abiDirectoryPath = abiFolderPath || DEFAULT_ABI_FOLDER_PATH;
```

Comment:
DEFAULT_ABI_FOLDER_PATH and DEFAULT_BIN_FOLDER_PATH are guarded (__NOT_FOR_WEB__BEGIN__). Should we check that if they are undefined and abiFolderPath, binFolderPath are not passed and raise error?

Reply:
Makes sense. However I am still not sure how can DEFAULT_ABI_FOLDER_PATH, DEFAULT_BIN_FOLDER_PATH can be undefined. 

Having said that once openst-contracts npm package are integrated, we will not need path dependencies .

Reply:
@abhayks1 can you please expand on that? I cannot follow.